### ISSUE TITLE
Why using bilinear interpolation for label ?

### ISSUE THREAD

**Issue by joihn** [lxasqjc/Deformation-Segmentation#23]
Hello,
Thanks for this repo. 

About this line:
https://github.com/lxasqjc/Deformation-Segmentation/blob/8c8278db784888ace138d4e4e9f9b6cbadae0dd7/models/models.py#L529
`y_sampled = F.grid_sample(y.float().unsqueeze(1), grid_y).squeeze(1)`

by default, `F.grid_sample` will use bilinear interpolation, 
which I think will corrupt the label (for example, `1:cyclist`, `2:car`, after interpolation, `1.50 : nothing = corrupted label`). 

Is there a reason you are not using `nearest` interpolation ?

**Comment by lxasqjc** [comment]
the bilinear interpolation of F.grid_sample is for the sampling coordinates, not for pixel value

**Comment by joihn** [comment]
You are telling me `y` would contain **coordinate** and not pixel value=class_label  ? 
However this line seems to indicate y contain class_label, which seems to contradict the above
https://github.com/lxasqjc/Deformation-Segmentation/blob/8c8278db784888ace138d4e4e9f9b6cbadae0dd7/models/models.py#L419

**Comment by joihn** [comment]
https://github.com/lxasqjc/Deformation-Segmentation/blob/8c8278db784888ace138d4e4e9f9b6cbadae0dd7/models/models.py#L419

However, this line seems to indicate that y contain pixel_value = class_label, not coordinate as you said , what do you think ?

**Comment by lxasqjc** [comment]
please refer to [TORCH.NN.FUNCTIONAL.GRID_SAMPLE](https://pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html#torch.nn.functional.grid_sample)
